internal/commands/accesslist/delete.go

```diff
@@ -0,0 +1,195 @@
+package accesslist
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cli/user"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/10gen/realm-cli/internal/utils/flags"
+
+	"github.com/AlecAivazis/survey/v2"
+)
+
+const (
+	headerDeleted = "Deleted"
+	headerDetails = "Details"
+)
+
+var (
+	deleteTableHeaders = []string{headerAddress, headerComment, headerDeleted, headerDetails}
+)
+
+type deleteAllowedIPOutputs []deleteAllowedIPOutput
+
+type deleteAllowedIPOutput struct {
+	allowedIP realm.AllowedIP
+	err       error
+}
+type deleteInputs struct {
+	cli.ProjectInputs
+	Addresses []string
+}
+
+// CommandMetaDelete is the command meta for the `accesslist delete` command
+var CommandMetaDelete = cli.CommandMeta{
+	Use:         "delete",
+	Display:     "accesslist delete",
+	Description: "Delete an IP address or CIDR block from the Access List of your Realm app",
+	HelpText: `Removes an existing entry from the Access List of your Realm app. You will be
+prompted to select an IP address or CIDR block if none is provided in the
+initial command.`,
+}
+
+// CommandDelete is the `accesslist delete` command
+type CommandDelete struct {
+	inputs deleteInputs
+}
+
+// Flags is the command flags
+func (cmd *CommandDelete) Flags() []flags.Flag {
+	return []flags.Flag{
+		cli.AppFlagWithContext(&cmd.inputs.App, "to remove an entry in its Access List"),
+		cli.ProjectFlag(&cmd.inputs.Project),
+		cli.ProductFlag(&cmd.inputs.Products),
+		flags.StringSliceFlag{
+			Value: &cmd.inputs.Addresses,
+			Meta: flags.Meta{
+				Name: "ip",
+				Usage: flags.Usage{
+					Description: "Specify the IP address or CIDR block that you would like to delete",
+				},
+			},
+		},
+	}
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Handler is the command handler
+func (cmd *CommandDelete) Handler(profile *user.Profile, ui terminal.UI, clients cli.Clients) error {
+	app, err := cli.ResolveApp(ui, clients.Realm, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	allowedIPs, err := clients.Realm.AllowedIPs(app.GroupID, app.ID)
+	if err != nil {
+		return err
+	}
+
+	selected, err := cmd.inputs.resolveAllowedIP(ui, allowedIPs)
+	if err != nil {
+		return err
+	}
+
+	if len(selected) == 0 {
+		ui.Print(terminal.NewTextLog("No IP addresses or CIDR blocks to delete"))
```

Comment:
I would check this one by @kpham1231 first...
```suggestion
		ui.Print(terminal.NewTextLog("No IP addresses nor CIDR blocks to delete"))
```

Reply:
also, this seems to me better suited as:

```go
return errors.New("No IP addresses nor CIDR blocks to delete")
```

Reply:
Would this necessarily be an error though? For example, if the user decided not to "select" any addresses, this would be output as a text log rather than an error. Fine either way though, just want to keep it consistent with how we're handling `secrets` as well (outputting it as a text log).

Reply:
intuitively, yes, i would think an error (and it was more than likely just oversight, rather than intentional, back in the secrets days).  at the end of the day it's messaging to the user, and I don't think prefixing `"no allowed IPs to delete" with `"accesslist delete failed: "` is the worst thing (for one, why run it if you want nothing to be deleted?  and also, that leaves us with a consistent/single CLI output for the case when there are no secrets to delete, either if none were selected or none were returned from the api, etc)

Reply:
> I would check this one by @kpham1231 first...

I would go with the first one ("No IP Addresses or CIDR blocks to delete")—while both are grammatically correct, the second one sounds a bit weird to me for some reason